src/front-end/typescript/lib/index.ts

```diff
@@ -177,3 +179,22 @@ export function getSignInUrl(
 export function fileBlobPath(file: Pick<FileRecord, "id">) {
   return prefixPath(`api/files/${file.id}?type=blob`);
 }
+
+/**
+ * Intersperse React elements with a separator.
+ *
+ * @param collection
+ * @param fn - callback to transform iteratee to a React element
+ * @param separator - element placed between collection members
+ * @returns List of React elements
+ */
+export function intersperse<A>(
+  collection: A[],
+  fn: (a: A) => string | ReactElement,
+  separator: string | ReactElement
+) {
+  return map(collection, (val: A, index: number) => [
+    ...(index > 0 ? [separator] : []),
+    fn(val)
+  ]);
+}
```

Comment:
It looks like this function is missing a return type signature, and returns a `ReactElement[][]`. The 2D array might be unintentional? If so, you would need to use a `fold` or `reduce` instead of `map`.

I also recommend decoupling this from React by implementing a general `intersperse<A>(collection: A[], separator: A): A[]` in `shared/lib/index.ts` and wrapping that function for a more specific React version in `front-end/lib/index.ts`: `intersperse(collection: Array<string | ReactElement>, separator: string | ReactElement): ReactElement[]`.

In functional programming libraries, the "mapping" functionality is left to `map` or `fmap`, and `intersperse` just focuses on adding the separator to an existing list, hence the omission of the `fn` parameter above, but will leave it with you on how you ultimately want to implement this helper.